### DISCUSSION TITLE
How to move theming service out of preferences to frontend based state management

### DISCUSSION THREAD

**Discussion by najaarv** [eclipse-theia/theia#7719] [General]
I have a custom extension where I have cleaned up most preferences and registered Themes as a new standalone Menu Item where I have two options-Light/Dark which is still based on `Themeservice` from `@theia/core/lib/browser/theming`. It looks like this:

![theme](https://user-images.githubusercontent.com/59064660/80842356-ff65f180-8c1e-11ea-85bd-ad95132b88b9.PNG)

Now, I have started a single instance of theia using yarn start from browser-app directory and hosted it on a server. Multiple users can access the hosted URL and they all could open different widgets and they all have different workspaces. Now the problem is if one of the users changes the theme from the app URL/web page on his browser, the theming in all other users' workspaces changes. Why does this happen? Is the theme value stored as part of the backend application state? I'm pretty sure that's the case because even if I change the theme in browser incognito mode, it is persisted and the last selected theme is brought back up on the next browser reload. If so, what else is part of the backend state? I don't really need other preferences except theming, so how can I move this theme service logic to frontend based state management?

If that is not possible, does every user need to have their own application/backend processes running so that their theming preferences are unique and not shared? Can anyone help with the steps to do that?

**Comment by vince-fugnitto** [comment]
@najaarv the current theme is stored in the settings (user, workspace, or folder).
From what I can tell about your description, the user settings are updated and since everyone is sharing the instance they are getting the update as well when connected.

You may want to see previous discussions about multi-tenancy:
- [spectrum/is-theia-multi-tenant](https://spectrum.chat/theia/dev/is-theia-multi-tenant-in-any-way~e92b1c24-e086-46c4-a35d-00a7ff951f18)

**Comment by akosyakov** [comment]
@najaarv You will need to customize `UserStorageService` to store user preference differently according to your definition of a user identity. For workspace preferences, you will need to give them different workspaces.
